packages/core/server.py

```
@@ -116,18 +138,26 @@ def run_skill(self, name: str, args: Dict[str, Any]) -> Dict[str, Any]:
             if proc.returncode != 0:
-                return {"error": proc.stderr.strip() or "Skill failed"}
+                logger.error("Skill execution failed", extra={"skill_name": name, "returncode": proc.returncode, "stderr": proc.stderr})
+                raise SkillExecutionException(name, proc.stderr.strip() or "Skill failed")
 
             output = proc.stdout.strip()
             try:
-                return json.loads(output)
+                result = json.loads(output)
+                logger.info("Skill executed successfully", extra={"skill_name": name})
+                return result
             except:
```

Ruff (0.14.10)

Do not use bare `except`

(E722)